src/Lykke.Service.IroncladDecorator/Controllers/CallbackController.cs

```diff
@@ -62,26 +65,32 @@ public async Task<IActionResult> SigninCallback()
                 _log.Warning(SessinNotExistMessage);
                 return BadRequest(SessinNotExistMessage);
             }
-           
-            var authCode = HttpContext.Request.Query["code"];
 
-            var tokens = await GetTokens(authCode);
+            var authenticationResponse = HttpContext.GetOpenIdConnectMessage();
+            
+            var authCode = authenticationResponse.Code;
+            
+            _openIdValidators.ValidateWebClientAuthenticationResponse(userSession, authenticationResponse);
```

Comment:
What will happen if validation will fail?

Reply:
An Exception would be thrown.